Fix crash during TaskScheduler creation

sobkas opened the pull request:
Fixes run when compiled with gcc higher than 10

Tested it on gcc-14 and binary worked.
But need more tests.

sobkas committed:
gcc: Fix compilation with newer versions of gcc

Was able to produce working binary using gcc-14

Xottab-DUTY commented on src/xrCore/Threading/TaskManager.hpp:
```diff
@@ -78,6 +78,7 @@ class XRCORE_API TaskManager final
 public:
     void RegisterThisThreadAsWorker();
     void UnregisterThisThreadAsWorker();
+    void MakeThreads();
```
```suggestion
    void SpawnThreads();
```

Xottab-DUTY commented on src/xrCore/xrCore.cpp:
```diff
@@ -274,6 +274,7 @@ void xrCore::Initialize(pcstr _ApplicationName, pcstr commandLine, bool init_fs,
         Msg("\ncommand line %s\n", Params);
         _initialize_cpu();
         TaskScheduler = xr_make_unique<TaskManager>();
+        TaskScheduler->MakeThreads();
```
```suggestion
        TaskScheduler->SpawnThreads();
```

Xottab-DUTY commented on src/xrCore/Threading/TaskManager.cpp:
```diff
@@ -230,6 +219,20 @@ TaskManager::~TaskManager()
     }
 }
 
+void TaskManager::MakeThreads()
```
```suggestion
void TaskManager::SpawnThreads()
```

Xottab-DUTY committed:
Renamed MakeThreads -> SpawnThreads

Changed wording to follow the terminology in Threading utility (which has functions named RunThread and SpawnThread)

Xottab-DUTY committed:
Update TaskManager.cpp

Primary thread should be registered as worker right from the start.
CalcIterations() outputs are used in worker threads only.
Moved some logic back and forth according to that.
Moved the SpawnThreads functions itself to make it consistent with the destructor (thread spawn first, thread despawn second)

Xottab-DUTY committed:
xrCore/Threading/TaskManager.hpp: moved SpawnThreads to be near constructor
